### ISSUE TITLE
Feature req: Jump-to-Letter in List Views

### ISSUE THREAD

**Issue by GoogleCodeExporter** [peteaky/android-xbmcremote#53]
```
In XBMC there's a nice feature that allows the user to jump to entries 
starting with a specific letter in (all, I guess) list views.
This is done by holding down <shift> and pressing the desired letter key.

A similar feature for xbmc-remote would be nice - especially when long 
lists are involved (lots of albums, etc.)

Attached is a patch that enables this feature.
Currently it only reacts to ascii letters [a-zA-Z], but with some love and 
care, it could be extended to allow numerics and even UTF-8 special 
characters.

Let me know what you think!
I'm by no means a GUI code wiz, so what I propose may not be the best way 
to implement this. It works, though... :]

:O) Mikkle

```

Original issue reported on code.google.com by `mukkenb...@gmail.com` on 27 Oct 2009 at 3:29

Attachments:
- [android-xbmc-remote-jump-to-letter-in-listviews.diff](https://storage.googleapis.com/google-code-attachments/android-xbmcremote/issue-53/comment-0/android-xbmc-remote-jump-to-letter-in-listviews.diff)

**Comment by GoogleCodeExporter** [comment]
```
Thought about gestures for android 1.6 to do exactly this. Will have a look at 
it.
Thanks so far, great to have your patches :)
```

Original comment by `till.ess...@googlemail.com` on 27 Oct 2009 at 9:52
- Changed state: **Accepted**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Wow, yeah, of course, gestures!
Only read the 1.6 briefs just now.

Gestures are definitely the way to go, but I guess my old-school input-method 
would 
still be applicable for pre-donut devices...

:O) Mikkle

```

Original comment by `mukkenb...@gmail.com` on 27 Oct 2009 at 11:51

**Comment by GoogleCodeExporter** [comment]
```
Yeah, if updating all devices takes as long as the update to cupcake it will 
take
some time to be sure everyone uses donut.
I think I'll have a look into your patch tomorrow.
```

Original comment by `till.ess...@googlemail.com` on 28 Oct 2009 at 12:02

**Comment by GoogleCodeExporter** [comment]
```
Totally love this patch, even though it only works with hardware keyboards... 
but I
don't know how this could be done with the virtual keyboard. Adding your patch 
to the
remote.
Thanks again mukkenborg
```

Original comment by `till.ess...@googlemail.com` on 28 Oct 2009 at 4:39
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Hmmm, reopened to be aware of the need of a virtual keyboard ;)
```

Original comment by `till.ess...@googlemail.com` on 28 Oct 2009 at 4:43
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Well, I tested it with virtual keyboard on the wife's Magic: press and hold the 
menu-
button, and the soft-keyboard will overlay the current view :)

:O) Mikkle
```

Original comment by `mukkenb...@gmail.com` on 28 Oct 2009 at 5:52

**Comment by GoogleCodeExporter** [comment]
```
Is that a Magic feature? Wont work on my G1...
```

Original comment by `till.ess...@googlemail.com` on 28 Oct 2009 at 6:14

**Comment by GoogleCodeExporter** [comment]
```
Dunno, really - I thought it was an android thing, and that's why I didn't 
bother to 
look into how to actually activate the virtual keyboard... I'll have a look 
around and 
see if I can find out if every android phone has a pop-up-keyboard-hot-key 
(which I 
hope) or if we need to activate the vk programatically...

:O) Mikkle
```

Original comment by `mukkenb...@gmail.com` on 28 Oct 2009 at 7:28

**Comment by GoogleCodeExporter** [comment]
```
Hmmm... Rt. Magic had the on-screen kbd from day one. G1 seems to have received 
it in 
the cupcake-upgrade... Research still on-giong...

:O) Mikkle
```

Original comment by `mukkenb...@gmail.com` on 28 Oct 2009 at 8:08

**Comment by GoogleCodeExporter** [comment]
```
I'll take a swipe at implementing something based on this sample:
http://developer.android.com/guide/samples/SoftKeyboard/index.html

:O) Mikkle
```

Original comment by `mukkenb...@gmail.com` on 28 Oct 2009 at 8:20

**Comment by GoogleCodeExporter** [comment]
```
Could have told you the thing with Android not having a virtual keyboard till 
cupcake ;)

Reimplementing a SoftKeyboard won't be the best idea, there have to be a way to 
just
open the normal keyboard (if it is possible on the magic it is everywhere) and 
use this. 
Even a search could be implemented with your work. Wont be a big deal from what 
I read.
```

Original comment by `till.ess...@googlemail.com` on 28 Oct 2009 at 9:02

**Comment by GoogleCodeExporter** [comment]
```
Yeah, you're right. I'd also prefer to open the default vk.
```

Original comment by `mukkenb...@gmail.com` on 29 Oct 2009 at 6:34

**Comment by GoogleCodeExporter** [comment]
```
Working solution attached :]
works on the emulator, that is... Haven't tried it on the Magic yet.
- There may be some problems with the magic already interpreting 
long-menu-press, but 
there ought to be a way around that.
On the emulator at least, the kbd can be dismissed by pressing the back-key, 
but I've 
added dismissal of the kbd when the menu-key is pressed (if the kbd is shown, 
of 
course)
My definition of a long-press is - in the patch - a repeat-count of 15, but 
that's 
just an arbitrary number - could be anything, really.

:O) Mikkle
```

Original comment by `mukkenb...@gmail.com` on 29 Oct 2009 at 3:24

**Comment by GoogleCodeExporter** [comment]
```
Replacement patch to enable software keyboard (deleted my previous uploaded 
patch - 
too much weird stuff in it).

Current patch activates softkeys by long-pressing the menu key. This is default 
behaviour on the HTC Magic, and it seems to be the recommended way of 
activating the 
softkbd even if it's not the most intuitive way.

The patch works in all music views and in video and pictures view (that's where 
it's 
needed, I guess).

Incidentially, I just realised that my original jump-to-letter-patch does not 
work in 
the FileListView (video, pictures, and music-file-view). The default android 
"incremental search" seems to be in effect in those views, though.
Your thoughts?

:O) Mikkle
```

Original comment by `mukkenb...@gmail.com` on 30 Oct 2009 at 12:46

Attachments:
- [android-xbmc-remote-issue53-software-keyboard-v2.diff](https://storage.googleapis.com/google-code-attachments/android-xbmcremote/issue-53/comment-14/android-xbmc-remote-issue53-software-keyboard-v2.diff)

**Comment by GoogleCodeExporter** [comment]
```
Be aware that the "video filelist" - with black background - differs from the 
file
mode tab in the music library. Make it work there, the old one will be wiped 
out soon.

Otherwise is there anything to do? Otherwise close the issue please. The 
long-press
for softkeyboard should be in the wiki somewhere (or on the main page)
```

Original comment by `phree...@gmail.com` on 3 Nov 2009 at 2:07

**Comment by GoogleCodeExporter** [comment]
```
closed by rev #268
```

Original comment by `mukkenb...@gmail.com` on 7 Nov 2009 at 10:04
- Changed state: **Done**
